File: Android_Studio/Java/Alcool_ou_Gasolina/app/src/main/java/com/example/alcoolougasolina/MainActivity.java
package com.example.alcoolougasolina;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity
{
    private TextInputEditText editPrecoAlcool, editPrecoGasolina;
    private TextView textoResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editPrecoAlcool = findViewById(R.id.editPrecoAlcool);
        editPrecoGasolina = findViewById(R.id.editPrecoGasolina);
        textoResultado = findViewById(R.id.textoResultado);
    }

    public void calcularPreco(View view)
    {
        // recuperar valores digitados
        String precoAlcool = editPrecoAlcool.getText().toString();
        String precoGasolina = editPrecoGasolina.getText().toString();

        // validar os campos digitados
        Boolean camposValidados = validarCampos(precoAlcool, precoGasolina);
        if (camposValidados)
        {
            resultado(precoAlcool, precoGasolina);
        }
        else
        {
            textoResultado.setText("Preencha os preços primeiro!");
        }
    }

    public Boolean validarCampos(String pAlcool, String pGasolina)
    {
        Boolean camposValidados = true;

        if(pAlcool == null || pAlcool.equals(""))
        {
            camposValidados = false;
        }
        else if (pGasolina == null || pGasolina.equals(""))
        {
            camposValidados = false;
        }

        return camposValidados;
    }

    public void resultado(String precoAlcool, String precoGasolina)
    {
        // convertendo string para números
        Double valorAlcool = Double.parseDouble( precoAlcool);
        Double valorGasolina = Double.parseDouble( precoGasolina);

        // cálculo de menor preço
        Double resultado = valorAlcool/valorGasolina;

        if (resultado >= 0.7)
        {
            textoResultado.setText("É melhor utilizar Gasolina!");
        }
        else
        {
            textoResultado.setText("É melhor utilizar Álcool!");
        }
    }
}

File: Android_Studio/Kotlin/Alcool_ou_Gasolina/app/src/main/java/com/example/alcoolougasolina/MainActivity.kt
package com.example.alcoolougasolina

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity()
{

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun btCalcular(@Suppress("UNUSED_PARAMETER")view: View)
    {
        // Recupera valores digitados
        //val precoAlcool = findViewById<View>(R.id.edit_preco_alcool) as TextView
        // Apenas com apply plugin: 'kotlin-android-extensions' em Gradle Scripts => build.gradle
        val precoAlcool = edit_preco_alcool.text.toString()
        val precoGasolina = edit_preco_gasolina.text.toString()

        val validaCampos = validarCampos(precoAlcool, precoGasolina)
        if (validaCampos == true)
        {
            calcularMelhorPreco(precoAlcool, precoGasolina)
        }
        else
        {
            text_resultado.setText("Preencha os campos vazios!")
        }
    }

    fun validarCampos(precoAlcool: String, precoGasolina: String) : Boolean
    {
        var camposValidados: Boolean = true

        if (precoAlcool.equals("")) //precoAlcool == null (null sempre será falso)
        {
            camposValidados = false
        }
        else if(precoGasolina.equals("")) //precoGasolina == null (null sempre será falso)
        {
            camposValidados = false
        }

        return camposValidados
    }

    fun calcularMelhorPreco(precoAlcool: String, precoGasolina: String)
    {
        // Converter valores strings para números
        val valorAlcool = precoAlcool.toDouble()
        val valorGasolina = precoGasolina.toDouble()

        /* Faz cálculo (precoAlcool/precoGasolina)
            * Se resultado >= 0.7 melhor utilizar Gasolina
            * senão melhor utilizar Álcool
         */
        val resultadoPreco = valorAlcool / valorGasolina

        if (resultadoPreco >= 0.7)
        {
            text_resultado.setText("Melhor utilizar Gasolina!")
        }
        else
        {
            text_resultado.setText("Melhor utilizar Álcool!")
        }
    }
}
